docker/gating-test-images/build.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+#
+# Script that builds and pushes to the internal docker registry a postgresql image with a test data sql dump,
+# and a candlepin image with an installed candlepin based on the current version of the stage environment.
+# - Intermediate, temporary images are built for efficiency (temp-base-cp)
+#   and loading/dumping of test data (temp-cp).
+# - All local images are removed after the push.
+
+retry() {
+    local -r -i max_attempts="$1"; shift
+    local -r name="$1"; shift
+    local -r cmd="$@"
+    local -i attempt_num=1
+    echo -n "Waiting for $name to be dumped..."
+    until ${cmd}
+    do
+        if (( attempt_num == max_attempts ))
+        then
+            echo "Attempt $attempt_num failed and there are no more attempts left!"
+            return 1
+        else
+            echo -n '.'
+            sleep $(( attempt_num++ ))
+        fi
+    done
+    echo
+}
+
+evalrc() {
+    if [ "$1" -ne "0" ]; then
+        echo "$2"
+        exit $1
+    fi
+}
+
+# clean up leftover files from potential previous runs
+rm -f sql/dump.sql
+rm -f postgres/dump.sql
+
+echo "============ Building temporary base candlepin image ============ "
+# builds base centos image which installs candlepin dependencies & environment
+docker build --tag=temp_base_candlepin temp-base-cp/
+evalrc $? "temp_base_candlepin image build was not successful."
+
+echo "============ Building temporary candlepin image ============ "
+# builds image which deploys a temporary candlepin, loads test data, and then dumps them
+docker build --tag=temp_candlepin temp-cp/
+evalrc $? "temp_candlepin image build was not successful."
+
+echo "============ Deploying temporary candlepin and loading test data ============ "
+# Need to temporarily disable selinux to avoid denial when writing on the .sql:/db-data bind mount
+setenforce 0
+
+# need to be in a swarm in order to use the stack command
+docker swarm init
+
+# runs the previously build image against a postgres database in a temporary stack,
+# to load and then dump the test data in dump.sql
+docker stack deploy -c temp-cp/docker-compose.yml load_and_dump_stack
+evalrc $? "load_and_dump_stack stack deploy was not successful."
+
+echo "============ Waiting for data to be dumped... ============ "
+retry 30 "dump.sql" test -f sql/dump.sql
+evalrc $? "dump.sql did not get dumped in time. Exiting..."
+
+setenforce 1
```

Comment:
Not sure if this is important for the expected environment, but this actually turned my SELinux back on since I run with it not enforcing typically. If there's any reasonable possibility that this'll be run in such an environment, maybe we should store the current setting above, then only turn it back on if necessary.

Reply:
Fixed.